packages/vite/src/node/plugins/resolve.ts

```diff
@@ -546,7 +546,7 @@ function tryResolveFile(
           const resolved = resolvePackageEntry(file, pkg, targetWeb, options)
           return resolved
         } catch (e) {
-          if (e.code !== 'ENOENT') {
+          if (e.code !== 'ENOENT' && e.code !== 'EACCES') {
```

Comment:
If we don't have read permission to `file`, the condition at 537 will be false. So EACCES here means that `file + '/package.json'` exists but we don't have access?
If this is an issue in the user folder, throwing is right here I think.
@brillout would you add a test case to better understand this issue?